### ISSUE TITLE
How can I predict multiple steps for time-series data in Keras using LSTM?

### ISSUE THREAD

**Issue by whutyw** [fchollet/keras#2384]
I know that timeDistributedDense can predict sequence, but I want to output sequence after the input sequence have complete read, like machine translation (the 4th figure from left). Thanks!
![image](https://cloud.githubusercontent.com/assets/7909907/14609590/1060c3f8-05bd-11e6-8c33-69d056593eff.png)

**Comment by keunwoochoi** [comment]
Disclaimer: I haven't done it.
You need to use tokens for the beginning and the end of sentences such as `_START_` and `_END_`. In the training data, probably you should pad zeros after the `_END_` for every input sentences, and do the same before the `_START_` for every output sentences. At the end, all the input/output sentences would be the same length. `timeDistributedDense` should be used.
Then learn the RNN with the data.
In prediction, feed the input sentences with tokens and do prediction until the output sentence predict `_END_`.

I think it makes sense, but not sure it's the right approach. Probably someone can correct me?

**Comment by carlthome** [comment]
Yes, that would be a good solution. Karpathy's post that the figure comes from used start and end tokens in that manner. Basically you'll construct your learning like the fifth figure from the left. In Keras you could introduce a dummy value (something your data could never be, let's say -1 which isn't even in the ASCII table) and use that for padding and add a [Masking layer](http://keras.io/layers/core/#masking) to make sure the masked values aren't used in any calculations.

**Comment by keunwoochoi** [comment]
I stumble upon [this paper](http://arxiv.org/abs/1506.00019). Figure 12 illustrates how you should do.
